### DISCUSSION TITLE
[5.x] FR: Show entry type icon in matrix field when only one entry type is seleected

### DISCUSSION THREAD

**Discussion by ZapaNOR** [craftcms/cms#14458] [Ideas] [answered]
After some testing of matrix in matrix I have noticed that icons and name is not shown for blocks when there is only one available block type for a matrix field. When creating deeper structures with sections and columns, this gets a bit messy, and an icon + name to spot the columns and sections at a glance would help a lot.

![Screenshot 2024-02-23 at 11 43 38](https://github.com/craftcms/cms/assets/53942798/db99724e-ed5b-49ec-8c7b-9cb8da790d69)
![Screenshot 2024-02-23 at 11 54 37](https://github.com/craftcms/cms/assets/53942798/43554858-f7f4-405c-985a-99a220320605)

**Comment by mmikkel** [comment]
I don't know, but I'm guessing the assumption is that for inline-editable Matrix fields with a single entry type, displaying the entry type name + icon is going to be redundant (as all the nested entries will use the same type).

OPs use case proves that this assumption isn't necessarily true, and I'm sure there are other situations where showing the type name and icon for single-type fields is desirable (for example, a single-type Matrix field wedged in-between other Matrix fields in the same field layout). Personally I don't see the big point in ever hiding away the type name, and I think the potential to confuse or disorient authors by doing so will always be there – so personally I'd prefer if it was just always visible.

**Comment by brandonkelly** [comment, marked as answer]
Yeah this was intentional, but agree, the slight aesthetic improvement isn’t worth potentially added confusion. Resolved for the next release (8ca30fcf7b536e2f9e1a418479692b6039b178af).

**Reply by brandonkelly** [reply]
5.0.0-beta.4 is out with that change.
